pkg/serverless/trace/cold_start_span_creator.go

```diff
@@ -35,45 +33,90 @@ const (
 var functionName = os.Getenv(functionNameEnvVar)
 
 type ColdStartSpanCreator struct {
-	executionContext *executioncontext.ExecutionContext
-	traceAgent       *ServerlessTraceAgent
-	createSpan       *sync.Once
+	TraceAgent            *ServerlessTraceAgent
+	CreateSpan            *sync.Once
+	LambdaSpanChan        chan *pb.Span
+	InitDurationChan      chan float64
+	SyncSpanDurationMutex *sync.Mutex
```

Comment:
Just a thought. One thing you can do is embed the mutex instead of adding it as a field. This will let you call the Lock and Unlock methods on the `ColdStartSpanCreator` itself. This is a pretty common go pattern. For example,

```
type ColdStartSpanCreator struct {
    sync.Mutex
    TraceAgent    *ServerlessTraceAgent
    ...
}

func (c *CodeStartSpanCreator) doIt() {
    c.Lock()
    defer c.Unlock()
    ...
}
```

Additionally, you do not need to make the embed a pointer, instead just leave it as the `sync.Mutex` alone. This is okay because it looks like all methods on the `ColdStartSpanCreator` have pointer receivers. The benefit of this is that when you create a `ColdStartSpanCreator`, you then do not need to instantiate the mutex directly. From the go docs,

> A Mutex is a mutual exclusion lock. The zero value for a Mutex is an unlocked mutex.